src/Domain/DomainHelpers.hpp

```diff
@@ -0,0 +1,636 @@
+  static std::vector<Direction<VolumeDim>> get_directions_along_face(
+      const std::array<size_t, two_to_the(VolumeDim - 1)>& face_local_corners);
+
+  /// Takes a Direction in block 1 and returns the corresponding Direction
+  /// in block 2.
+  static Direction<VolumeDim> mapped(
+      size_t block_id1, size_t block_id2,
+      const Direction<VolumeDim>& dir_in_block1,
+      const std::vector<std::array<size_t, two_to_the(VolumeDim)>>&
+          corners_of_all_blocks);
+
+  /// Given ids of two blocks, returns the Directions that correspond
+  /// to each block.
+  static std::pair<std::array<Direction<VolumeDim>, VolumeDim>,
+                   std::array<Direction<VolumeDim>, VolumeDim>>
+  obtain_correspondence_between_blocks(
+      size_t block_id1, size_t block_id2,
+      const std::vector<std::array<size_t, two_to_the(VolumeDim)>>&
+          corners_of_all_blocks);
+
+  /// Given Directions in block1 and their corresponding Directions
+  /// in block2, this function first repackages them into arrays and
+  /// then makes a std::pair of the arrays.
+  static std::pair<std::array<Direction<VolumeDim>, VolumeDim>,
+                   std::array<Direction<VolumeDim>, VolumeDim>>
+  create_correspondence_between_blocks(
+      const Direction<VolumeDim>& direction_to_neighbor_in_self_frame,
+      const Direction<VolumeDim>& direction_to_self_in_neighbor_frame,
+      const std::vector<Direction<VolumeDim>>&
+          alignment_of_shared_face_in_self_frame,
+      const std::vector<Direction<VolumeDim>>&
+          alignment_of_shared_face_in_neighbor_frame);
+
+  /// \returns the RMS separation between two blocks in a domain via their
+  /// CoordinateVolumeDims
+  /// This can be used to confirm they are physically connected.
+  template <typename Map1, typename Map2>
+  static double physical_rms_separation(
+      const Map1& map_of_block1, const Map2& map_of_block2,
+      const std::array<size_t, two_to_the(VolumeDim)>& corners_of_block1,
+      const std::array<size_t, two_to_the(VolumeDim)>& corners_of_block2);
+
+  /// When setting up periodic-like boundary conditions, the user specifies
+  /// the corners of the two faces to be identified.
+  /// \requires that the two faces correspond to the same block.
+  /// \returns the corresponding id of the block whose faces are to
+  /// be identified.
+  static size_t find_block_id(
+      const std::array<size_t, two_to_the(VolumeDim - 1)>& global_corners_face1,
+      const std::array<size_t, two_to_the(VolumeDim - 1)>& global_corners_face2,
+      const std::vector<std::array<size_t, two_to_the(VolumeDim)>>&
+          corners_of_all_blocks);
+
+  /// Sets up the BlockNeighbors using the corner numbering scheme
+  /// to deduce the correct neighbors and orientations. Does not set
+  /// up periodic-like boundary conditions.
+  template <typename Map>
+  static void set_internal_boundaries(
+      const std::vector<std::unique_ptr<Map>>& maps_of_all_blocks,
```

Comment:
I don't think so - also I've recently realized that since CoordinateMap isn't a base class this code will not work - `maps_of_all_blocks` will need to hold CoordinateMaps of different types

Reply:
Okay, you'll likely want to have it by a `std::tuple` and then use `tuple_fold` or `tuple_transform` to iterate over them.